src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -139,56 +125,31 @@ public void removePerson(Person key) {
      * @param sg the SuperGroup to be added into AddressBook.
      */
     public void addSuperGroup(SuperGroup sg) {
-        if (!superGroups.containsKey(sg.getName())) {
-            superGroups.put(sg.getName(), sg);
-        }
+        superGroups.add(sg);
     }
 
     public void deleteSuperGroup(SuperGroup sg) {
-        superGroups.remove(sg.getName());
+        superGroups.remove(sg);
     }
 
     /**
      * Gets SuperGroup based on group name.
      */
     public SuperGroup findSuperGroup(String name) {
-        return superGroups.get(name);
-    }
-
-    /**
-     * Returns true if SuperGroup exists.
-     */
-    public boolean hasSubGroup(SubGroup subGroup) {
-        requireNonNull(subGroup);
-        return subGroups.containsKey(subGroup.toString());
-    }
-
-    /**
-     * Adds SubGroup into the AddressBook.
-     * @param subGroup the SuperGroup to be added into AddressBook.
-     */
-    public void addSubGroup(SubGroup subGroup) {
-        requireNonNull(subGroup);
-        subGroups.put(subGroup.toString(), subGroup);
-    }
-
-    public void deleteSubGroup(SubGroup subGroup) {
-        subGroups.remove(subGroup.toString());
-    }
-
-    /**
-     * Gets SuperGroup based on group name.
-     */
-    public SubGroup findSubGroup(String name) {
-        return subGroups.get(name);
+        for (SuperGroup superGroup: superGroups) {
+            if (superGroup.getName().equals(name)) {
+                return superGroup;
+            }
+        }
+        return null;
     }
 
     /**
      * Returns true if SuperGroup exists.
      */
     public boolean hasSuperGroup(SuperGroup superGroup) {
         requireNonNull(superGroup);
-        return superGroups.containsKey(superGroup.getName());
+        return superGroups.contains(superGroup);
     }
```

Comment:
Same issue as above, maybe leave this to a lower level of abstraction in the UniqueList?

Reply:
It's referring to the use of SLAP above, just wait until the refactor is done I guess